### PR TITLE
Add validations to copyAPI Implementation

### PR DESCRIPTION
**Pull request by anuruddhal**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java [hunk 1]
```diff
@@ -620,9 +620,19 @@ public void updateCheckListItem(String apiId, String status, Map<String, Boolean
     public String createNewAPIVersion(String apiId, String newVersion) throws APIManagementException {
         String newVersionedId;
         LifecycleState lifecycleState;
+        if (StringUtils.isEmpty(newVersion)) {
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: better to do an empty check for apiId as well and throw similar error

Comment (anuruddhal) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: This is already tested at DAO layer using annotation.
https://github.com/wso2/carbon-apimgt/blob/C5/components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/ApiDAO.java#L45

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java [hunk 2]
```diff
@@ -620,9 +620,19 @@ public void updateCheckListItem(String apiId, String status, Map<String, Boolean
...
+            String errorMsg = "New API version cannot be empty";
+            log.error(errorMsg);
+            throw new APIManagementException(errorMsg, ExceptionCodes.PARAMETER_NOT_PROVIDED);
+        }
         try {
             API api = getApiDAO().getAPI(apiId);
             if (api != null) {
+                if (api.getVersion().equals(newVersion)) {
+                    String errMsg = "New API version cannot be same as the previous version";
```
Comment (prasa7) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: Shall we add context information to the log. 

Example : "New API version" + newVersion  +  "cannot be same as the previous version" for API + APIName

Comment (anuruddhal) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java]: Fixed in PR https://github.com/wso2/carbon-apimgt/pull/3843

**Pull request merged**